portal-api/src/v1/users/sign-in-link.service.js

```diff
@@ -17,37 +18,51 @@ class SignInLinkService {
   async createAndEmailSignInLink(user) {
     const { _id: userId, email: userEmail, firstname: userFirstName, surname: userLastName } = user;
 
-    const signInCode = this.#createSignInCode();
-    await this.#saveSignInCodeHashAndSalt({ userId, signInCode });
+    const signInToken = this.#createSignInToken();
+    await this.#saveSignInTokenHashAndSalt({ userId, signInToken });
 
     return this.#sendSignInLinkEmail({
-      signInLink: `${PORTAL_UI_URL}/login/sign-in-link?t=${signInCode}`,
+      signInLink: `${PORTAL_UI_URL}/login/sign-in-link?t=${signInToken}`,
       userEmail,
       userFirstName,
       userLastName,
     });
   }
 
-  #createSignInCode() {
+  async isValidSignInToken({ username, signInToken }) {
+    return new Promise((resolve, reject) => {
+      findByUsername(username, (error, user) => {
```

Comment:
Again, I dislike the callback structure we are using here. I'm tempted to depreciate `findByUsername` and make its implementation call our new `user repo` (pending how many extra failing tests this is and others thoughts on the matter)